src/QAT/qat/purr/backends/live_devices.py

```diff
@@ -36,7 +36,7 @@ def disconnect(self):
         if self.driver is not None:
             try:
                 self.driver.close()
-                self.driver = None
+                self._driver = None
```

Comment:
I agree this breaks the call cycle `disconnect()` <-> `driver.setter()` but I don't think the setter should be called here at all.

Any idea why it's being called ?

Reply:
Good point. :thinking:

Reply:
Discussed with relevant parties and for the current driver versions this is required for proper operation.

Reply:
can we add a comment to this effect? I'll certainly forget this tomorrow

Reply:
on the other hand, the setter disconnects the driver also to potential reference owners. It isn't an issue now but might be in the future ...

Reply:
Happy with the fix for the time being. For good measure, could we sweep at other usages of the setter and make sure to use the private field `_driver` directly from within `Instrument` ? I would like to keep the getter/setter is only for client structures